src/runtime/substitution.cc

```diff
@@ -841,7 +847,16 @@ bool GraphXfer::create_new_operator(OpX const *opx, Node &op) {
     case OP_CONCAT: {
       int axis;
       assert(opx->get_pm_constraint(PM_AXIS, axis));
-      op = model->get_or_create_concat_node(opx->inputs.size(), inputs, axis);
+
```

Comment:
Why can't we just pass in `inputs` here as opposed to doing a rather verbose copy into `_inputs`?

Reply:
`inputs` is of type `int[]` and we need a `vector<int>` for the template functions that convert `ParallelTensor` to `ParallelTensorShape`. That being said, I've replaced the loop above with `
        std::vector<ParallelTensor> _inputs (std::begin(inputs), std::end(inputs));`  for something more elegant.